src/EnergyPlus/WeatherManager.cc

```diff
@@ -2317,27 +2318,19 @@ namespace WeatherManager {
             }
         }
 
-        // if using epw precipitation, throw warning
-        if (state.dataWaterData->RainFall.ModeID == DataWater::RainfallMode::EPWPrecipitation) {
-            if ((state.dataEnvrn->LiquidPrecipitation == 0) && (state.dataEnvrn->IsRain)) {
-                state.dataEnvrn->LiquidPrecipitation = 1.5 / 1000.0;
-                ShowRecurringWarningErrorAtEnd(state,
-                                               "Rain flag is on but precipitation depth in the weather file is missing or zero. Setting "
-                                               "precipitation depth to 1.5 mm for this hour.",
-                                               state.dataWaterData->PrecipOverwrittenByRainFlag);
-            }
-        }
-
         WaterManager::UpdatePrecipitation(state);
 
         state.dataEnvrn->TotalCloudCover = state.dataWeatherManager->TodayTotalSkyCover(state.dataGlobal->TimeStep, state.dataGlobal->HourOfDay);
         state.dataEnvrn->OpaqueCloudCover = state.dataWeatherManager->TodayOpaqueSkyCover(state.dataGlobal->TimeStep, state.dataGlobal->HourOfDay);
 
         if (state.dataWeatherManager->UseRainValues) {
-            state.dataEnvrn->IsRain = state.dataWeatherManager->TodayIsRain(
-                state.dataGlobal->TimeStep, state.dataGlobal->HourOfDay); //.or. LiquidPrecipitation >= .8d0)  ! > .8 mm
-        } else {
-            state.dataEnvrn->IsRain = false;
```

Comment:
@yujiex Looks like this else was deleted? Is this covered elsewhere?

Reply:
In the header file, the IsRain is initialized to false. So I was thinking if I don't do anything, it should be false.

Reply:
But `UseRainValues` can change with each environment, so it needs to be reset at least once per environment.

Reply:
Right, I'll add it back.